File: Data Structures & Algorithms/maximum-score-after-splitting-a-string/submission-0.cpp
class Solution {
public:
    int maxScore(string s) {
        int rightOnes = 0;

        for (char c : s)
            if (c == '1')
                rightOnes++;

        int leftZeros = 0;
        int res = 0;

        for (int i = 0; i < s.size() - 1; i++) {
            if (s[i] == '0')
                leftZeros++;
            else
                rightOnes--;

            res = max(res , leftZeros + rightOnes);
        }

        return res;
    }
};

File: Data Structures & Algorithms/max-consecutive-ones/submission-0.py
class Solution:
    def findMaxConsecutiveOnes(self, nums: List[int]) -> int:
        l = len(nums)
        ones = 0 
        res = 0
        for i in range(l)  :
            if(nums[i] == 1) :
                ones += 1 
            if(nums[i] == 0) : 
                ones = 0 
            res = max(res , ones)

        return res